### PR TITLE
Feature/boas 1045 s51 publish

### PR DESCRIPTION
**Pull request by HadrianHughesKC**
## Describe your changes

* feat(api/applications): add new S51 advice topic to topics file
* refactor(api/applications): create a service method for publishing S51
* chore(api/applications): remove adviceReference from S51 advice event schema
We don't currently store a value like this
* feat(api/applications): build and publish S51 advice publish events

Will need to test this in dev as it relies on service bus.

## Issue ticket number and link

[BOAS-1045](https://pins-ds.atlassian.net/browse/BOAS-1045)

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have double checked this work does not include any hardcoded secrets or passwords
- [x] I have made corresponding changes to the documentation
- [x] I have provided details on how I have tested my code
- [x] I have referenced the ticket number above
- [x] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

[BOAS-1045]: https://pins-ds.atlassian.net/browse/BOAS-1045?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/api/src/server/applications/s51advice/s51-advice.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	method: s51Advice.enquiryMethod,
+	enquiryDate: s51Advice.enquiryDate,
+	enquiryDetails: s51Advice.enquiryDetails,
+	adviceGivenBy: s51Advice.adviser,
+	adviceDate: s51Advice.adviceDate,
+	adviceDetails: s51Advice.adviceDetails,
+	status: s51Advice.publishedStatus,
+	redactionStatus: s51Advice.redactedStatus,
+	// @ts-ignore
+	attachmentIds: s51Advice.S51AdviceDocument.map((doc) => doc.id)
```
Comment (chrisdalykinandcarta) [inline apps/api/src/server/applications/s51advice/s51-advice.js:40]: Can probs remove lots of this boilerplate with pick

File: apps/api/src/message-schemas/events/s51-advice.schema.json [hunk 2]
```diff
@@ -9,10 +9,6 @@
 			"type": "number",
 			"examples": [1234]
 		},
-		"adviceReference": {
```
Comment (chrisdalykinandcarta) [inline apps/api/src/message-schemas/events/s51-advice.schema.json:12]: How come we're removing adviceReference? I think this is a requirement unless something has changed

File: apps/api/src/server/applications/s51advice/s51-advice.js (hunk 1 above)
Comment (HadrianHughesKC) [reply apps/api/src/server/applications/s51advice/s51-advice.js:40]: Good shout 👍

File: apps/api/src/message-schemas/events/s51-advice.schema.json (hunk 2 above)
Comment (HadrianHughesKC) [reply apps/api/src/message-schemas/events/s51-advice.schema.json:12]: I can't see a column in the database to use for this. Looks like it's expecting something similar in format to the case reference.

Comment (HadrianHughesKC) [reply apps/api/src/message-schemas/events/s51-advice.schema.json:12]: Ah we do have a reference number which is an integer. Maybe the reference string needs to be constructed like `{caseReference}-{advice reference number}`

Comment (HadrianHughesKC) [reply apps/api/src/message-schemas/events/s51-advice.schema.json]: Ok I've just updated it. There is a function already that formats an advice reference.

**Review by chrisdalykinandcarta — APPROVED** [review]

**Pull request merged**